test/e2e_node/container_lifecycle_test.go

```diff
@@ -516,3 +517,1498 @@ var _ = SIGDescribe("[NodeConformance] Containers Lifecycle ", func() {
 		framework.ExpectNoError(results.HasRestarted(regular1))
 	})
 })
+
```

Comment:
I would like a node serial e2e test that adds a restartable init container, waits for it to be initialized, stops the kubelet, and then:

* causes the restartable init container to exit, starts the kubelet, verifies the restartable init container is started
* (if we have the machinery) removes all but the last init container, starts the kubelet, and verifies the pod is not re-initialized
* (if we have the machinery) removes all but the last init container (which should be restartable), starts the kubelet, and verifies the restartable init container is not restarted and than no other init containers are started.

I know we can do the first one, I assume we can do something slightly CRI specific to purge specific containers.

Reply:
That’s fine with me.

Reply:
made an issue https://github.com/kubernetes/kubernetes/issues/119014